vtysh/vtysh.c

```diff
@@ -2611,25 +2611,30 @@ DEFUN (vtysh_write_memory,
 	/* If integrated frr.conf explicitely set. */
 	if (want_config_integrated()) {
 		ret = CMD_WARNING_CONFIG_FAILED;
+
+		/* first attempt to use watchfrr if it's available */
+		bool used_watchfrr = false;
 		for (i = 0; i < array_size(vtysh_client); i++)
 			if (vtysh_client[i].flag == VTYSH_WATCHFRR)
 				break;
-		if (i < array_size(vtysh_client) && vtysh_client[i].fd != -1)
+		if (i < array_size(vtysh_client) && vtysh_client[i].fd != -1) {
+			used_watchfrr = true;
 			ret = vtysh_client_execute(&vtysh_client[i],
 						   "do write integrated",
 						   outputfile);
+		}
 
 		/*
-		 * If watchfrr returns CMD_WARNING_CONFIG_FAILED this means
-		 * that it could not write the config, but additionally
-		 * indicates that we should not try either
+		 * If we didn't use watchfrr, fallback to writing the config
+		 * ourselves
 		 */
-		if (ret != CMD_SUCCESS && ret != CMD_WARNING_CONFIG_FAILED) {
+		if (!used_watchfrr) {
 			printf("\nWarning: attempting direct configuration write without "
 			       "watchfrr.\nFile permissions and ownership may be "
 			       "incorrect, or write may fail.\n\n");
 			ret = vtysh_write_config_integrated();
```

Comment:
I'm really not comfortable with this text - it implies watchfrr is a necessary component to correct operation of FRR and this is simply not the case.  I think the printf should be removed.

Reply:
Then open a PR for it? How is that related to this pull request?

Reply:
Text is unrelated to PR, removal or change of it should be addressed in an own PR.